src/include/kdb.h.in

```diff
@@ -193,7 +198,8 @@ ssize_t keySetString(Key *key, const char *newString);
 ssize_t keyGetBinary(const Key *key, void *returnedBinary, size_t maxSize);
 ssize_t keySetBinary(Key *key, const void *newBinary, size_t dataSize);
 
-
+int keyLock (Key * key, option_t what);
```

Comment:
We should probably add a new `typedef int lock_t` so that users have a better idea which options are supported. Ideally we would use a `typedef enum`, but I don't know how well that works with bitwise operations (especially in C++).

Reply:
good idea. especially as it was a separated enum before...

Reply:
I won't do this. But you are free to add a dedicated enum later.

Reply:
Oh and `..._t` is reserved by glibc:  `Names that end with ‘_t’ are reserved for additional type names. `

Reply:
@manuelm You already earlier stated that `_t` is reserved. You are free to fix this. It will be quite a huge change. I think it is a quite minor issue, although option_t specifically is really a quite generic name.

Reply:
The glibc also reserves all names starting with an underscore followed by an uppercase letter. I don't know whether that applies to struct names as well, or just to symbols that can be used without the `struct ` or `enum ` prefix, but if it does apply to struct names `struct _Key` et al are also affected.

Reply:
But please do it in a different PR :wink: :

Reply:
> The glibc also reserves all names starting with an underscore followed by an uppercase letter.

This is also reserved by ANSI/ISO C.

> but if it does apply to struct names struct _Key et al are also affected.

As _Key is only used internally, I do not think it will be a problem for users of Elektra. Even if _Key exists in glibc, how likely is it that someone tries to pass this stuff to Elektra's `key*` functions?